src/background/Wallet/Wallet.ts

```diff
@@ -1728,30 +1842,119 @@ class PublicController {
+  }>) {
+    const currentAddress = this.wallet.ensureCurrentAddress();
+    if (!this.wallet.allowedOrigin(context, currentAddress)) {
+      throw new OriginNotAllowed();
+    }
+    const searchParams = new URLSearchParams({
+      origin: context.origin,
+      transaction: txBase64, // JSON.stringify(transaction),
+      ecosystem: 'solana',
+      method,
+    });
+    if (clientScope) {
+      searchParams.append('clientScope', clientScope);
+    }
+
+    return new Promise((resolve, reject) => {
+      this.safeOpenDialogWindow(context.origin, {
+        requestId: `${context.origin}:${id}`,
+        route: '/sendTransaction',
+        // height: isDeviceWallet ? 800 : undefined,
+        search: `?${searchParams}`,
+        tabId: context.tabId || null,
+        onResolve: (result: SolTransactionResponse) => {
+          console.log({ result });
+          resolve(result);
+        },
+        onDismiss: () => {
+          reject(new UserRejectedTxSignature());
+        },
+      });
+    });
+  }
+
+  async eth_requestAccounts(
+    {
+      context,
+      id,
+      params,
+    }: PublicMethodParams<[] | [{ nonEip6963Request?: boolean }] | null>,
+    opts: { ecosystem: 'ethereum' | 'solana' } = { ecosystem: 'ethereum' }
+  ) {
     if (debugValue && process.env.NODE_ENV === 'development') {
       // eslint-disable-next-line no-console
       console.log('PublicController: eth_requestAccounts', debugValue);
     }
     const currentAddress = this.wallet.readCurrentAddress();
-    if (currentAddress && this.wallet.allowedOrigin(context, currentAddress)) {
-      const { origin } = context;
-      emitter.emit('requestAccountsResolved', {
-        origin,
-        address: currentAddress,
-        explicitly: false,
-      });
-      // Some dapps expect lowercase to be returned, otherwise they crash the moment after connection
-      const result = [currentAddress.toLowerCase()];
-      if (debugValue && process.env.NODE_ENV === 'development') {
-        result.push(String(debugValue));
+    const addrIsEthereum = currentAddress && isEthereumAddress(currentAddress);
+    const addrIsSolana = currentAddress && isSolanaAddress(currentAddress);
+    if (opts.ecosystem === 'ethereum') {
+      invariant(!currentAddress || addrIsEthereum, 'addresss is not ethereum');
+    } else if (opts.ecosystem === 'solana') {
+      invariant(!currentAddress || addrIsSolana, 'addresss is not solana');
+    }
+    if (currentAddress)
+      if (
+        currentAddress &&
+        this.wallet.allowedOrigin(context, currentAddress)
+      ) {
+        const { origin } = context;
+        emitter.emit('requestAccountsResolved', {
+          origin,
+          address: currentAddress,
+          explicitly: false,
+        });
+        // Some ethereum dapps expect lowercase to be returned, otherwise they crash the moment after connection
+        const result = [
+          addrIsEthereum ? currentAddress.toLowerCase() : currentAddress,
```

Comment:
`normalizeAddress`?

Reply:
I think according to the existing comment we want to explicitly convert to lowercase here. `normalizeAddress` can be later refactored to normalize to checksum addresses instead